src/app/features/search/search.component.ts

```diff
@@ -12,7 +15,14 @@ import { GlobalErrorService } from '../global-error/global-error.service';
 export class SearchComponent {
   vehicleTechRecords$: Observable<Array<VehicleTechRecordModel>>;
 
-  constructor(private technicalRecordService: TechnicalRecordService, public globalErrorService: GlobalErrorService) {
+  constructor(private technicalRecordService: TechnicalRecordService, public globalErrorService: GlobalErrorService, 
+    private store: Store, private router: Router) {
+      this.store.pipe(select(selectQueryParams)).subscribe((params) => {
+        let vin = params['vin'];
+        if (vin !== null || vin !== undefined) {
```

Comment:
When loading the page vin will be undefined so it goes into the if block. This should be `&&` or better yet `if(vin)`.